### DISCUSSION TITLE
Why not WinGet instead of WSL>nvm or nvm-windows?

### DISCUSSION THREAD

**Discussion by TBemrose** [npm/feedback#868] [Documentation]
[npm Docs](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm#using-a-node-installer-to-install-nodejs-and-npm) says: 

> **We strongly recommend using a Node version manager like [nvm](https://github.com/nvm-sh/nvm) to install Node.js and npm.** We do not recommend using a Node installer, since the Node installation process installs npm in a directory with local permissions and can cause permissions errors when you run npm packages globally.

I'm not going to use multiple versions of node so the only reason I have to use a Node Version Manager would be to get node/npm installed properly. After carefully reading everything about npm and [nvm](https://github.com/nvm-sh/nvm), I see that in order to use nvm I'd have to use [WSL](https://learn.microsoft.com/en-us/windows/wsl/). For those keeping score, we're now _two_ superfluous steps removed from the goal, getting node/npm installed properly. Reading about WSL, I suspect that WSL>nvm will lock me into using node/npm inside a Linux environment, further complicating things. There's also [nvm-windows](https://github.com/coreybutler/nvm-windows) which points out that it's only _similar_ to nvm, so going that route would require remaining diligent whenever I read instructions for nvm.

At first I confused wget (for [installing nvm](https://github.com/nvm-sh/nvm#install--update-script) on Linux) with [WinGet](https://learn.microsoft.com/en-us/windows/package-manager/winget/) ([github](https://github.com/microsoft/winget-cli)) and spent time researching whether nvm's install.sh was cross-platform and could be run in PowerShell. I concluded it's not, hence WSL and nvm-windows. When I realized wget was not winget, I decided to see if I could just cut out all the middlemen steps.

```PowerShell
PS C:\Users> winget search npm; ""
Name            Id                    Version Match        Source
-----------------------------------------------------------------
Volta           Volta.Volta           1.1.1   Command: npm winget
Node.js Nightly OpenJS.NodeJS.Nightly 20.0.0  Command: npm winget
Node.js         OpenJS.NodeJS         19.6.0  Command: npm winget
pnpm            pnpm.pnpm             7.27.0  Tag: npm     winget
Node.js LTS     OpenJS.NodeJS.LTS     18.14.0 Tag: npm     winget
Nodist          Nodist.Nodist         0.9.1.0 Tag: npm     winget
ndm             720kb.ndm             1.2.0   Tag: npm     winget

PS C:\Users> winget show "Node.js LTS"
Found Node.js LTS [OpenJS.NodeJS.LTS]
Version: 18.14.0
Publisher: Node.js Foundation
Publisher Url: https://nodejs.org
Publisher Support Url: https://github.com/nodejs/help/issues
Author: OpenJS Foundation
Moniker: nodejs-lts
Description: Node.js is a JavaScript runtime built on Chrome"s V8 JavaScript engine. Node.js uses an event-driven, non-blocking I/O model that makes it lightweight and efficient. Node.js" package ecosystem, npm, is the largest ecosystem of open source libraries in the world.
Homepage: https://nodejs.org
License: Copyright Joyent, Inc. and other Node contributors. All rights reserved.
License Url: https://raw.githubusercontent.com/nodejs/node/v16.4.1/LICENSE
Release Notes Url: https://github.com/nodejs/node/releases/tag/v18.14.0
Tags:
  cross-platform
  foss
  javascript
  lts
  node
  nodejs
  npm
Installer:
  Installer Type: wix
  Installer Locale: en-US
  Installer Url: https://nodejs.org/dist/v18.14.0/node-v18.14.0-x64.msi
  Installer SHA256: 944eff6104be19d1dc24f3940ab365aa972c47ee2a6b7cfee49dd436e748bd99
```

In the end, `Installer Url` is the exact same installer I'd end up with from node.js' [download](https://nodejs.org/en/download/) page, so I've come full circle to npm Docs' warning against using a Node installer, which answers the first level of my title question. However, along this journey I've noticed that so much of all this documentation heavily assumes the reader is using Linux, so much so that it's often not clear when you're reading only applies to Linux. This unveils a second layer of my question:

**Does npm Docs' warning about Node installers putting npm in a directory with local permissions only apply to Linux?** Why not run PowerShell as administrator (of course) and use WinGet, or (seemingly equivalent) download the .msi and run it as administrator?

**Comment by ljharb** [comment]
If you're using more than one version of node across *all time*, you're using multiple versions of node. In the fullness of time, this means that *literally everyone* is always using multiple versions of node.

On windows, to use nvm, you'd need WSL 2, or cygwin, or git-bash.

**Reply by TBemrose** [reply]
That's a good insight, but it doesn't answer the remaining question: Do the Windows installers (installed with Admin permissions) have the issue warned about in the npm Docs, or is that only an issue with using the Linux installers (vs the suggested Linux installation method)?

**Reply by ljharb** [reply]
I believe the recommendation to use a version manager applies to any OS, including Windows, if that's what you mean.
